src/web/mod.rs

```diff
@@ -485,7 +487,12 @@ fn duration_to_str(init: DateTime<Utc>) -> String {
 fn redirect(url: Url) -> Response {
     let mut resp = Response::with((status::Found, Redirect(url)));
     resp.headers.set(Expires(HttpDate(time::now())));
+    resp
+}
 
+fn cached_redirect(url: Url, cache_policy: cache::CachePolicy) -> Response {
```

Comment:
That was actually an intentional choice for the first draft. 

My idea here: a `301` would be cache the redirect forever, also in the browser. 
This would make any change to our url-patterns harder to deploy, like the change to `/latest/` URLs instead of the specific version redirect. 

I believe `max-age=86400` was actually ignored because of the `301` status. 

I would love more opinions here from @rust-lang/docs-rs

Reply:
For now I adapted the logic in `rustdoc_redirector_handler` to match the current logic where we serve a permanent redirect below `/latest/`, while changing the max-age according to "permanent".

Reply:
The 301 behavior was introduced in https://github.com/rust-lang/docs.rs/pull/1760, with (partial) rationale in https://github.com/rust-lang/docs.rs/issues/1756. The idea was that Google would do a better job canonicalizing to the full `/xyz/latest/xyz/` URL if we used 301s. However, searching for more info now I see https://www.seroundtable.com/google-302-redirects-treated-301-redirects-31221.html which suggests Google doesn't much care about 301 vs 302, so I'm okay reverting to 302 for now.

Reply:
ok, reverted